tests/functional/object/mcg/ui/test_bucket_lifecycle_ui.py

```diff
@@ -0,0 +1,390 @@
+        )
+        assert backend_policy, "Failed to retrieve lifecycle policy from backend"
+
+        backend_rules = backend_policy.get("Rules", [])
+        our_rule = next(
+            (r for r in backend_rules if r["ID"] == initial_rule_name), None
+        )
+        assert our_rule, f"Rule {initial_rule_name} not found in backend policy"
+
+        assert "Expiration" in our_rule, "Expected Expiration in backend rule"
+        assert (
+            our_rule["Expiration"]["Days"] == new_days
+        ), f"Expected expiration days {new_days}, got {our_rule['Expiration']['Days']}"
+
+        logger.info(
+            f"Successfully edited lifecycle rule: {initial_rule_name} from {initial_days} to {new_days} days"
+        )
+
+    @ui
+    @tier2
+    @pytest.mark.jira("DFBUGS-2960")
+    @pytest.mark.polarion_id("OCS-6892")
+    def test_delete_lifecycle_rule(self, setup_ui_class_factory, mcg_obj):
+        """
+        Test deletion of lifecycle rule
+
+        Args:
+            setup_ui_class_factory: Pytest fixture for UI setup
+            mcg_obj: MCG object fixture for backend validation
+
+        Steps:
+        1. Navigate to an existing bucket's lifecycle rules page
+        2. Check for existing rules, create one if none exist
+        3. Delete the rule using kebab menu
+        4. Verify rule is removed from UI list
+        5. Verify rule is removed from backend
+        """
+        setup_ui_class_factory()
+        lifecycle_ui = BucketLifecycleUI()
+
+        if not self.created_buckets:
+            logger.warning("No buckets from previous tests, creating a new one")
+            lifecycle_ui, bucket_name = self._setup_bucket_and_navigate_to_lifecycle()
+        else:
+            bucket_name = self.created_buckets[0]
+            logger.info(f"Using existing bucket from previous tests: {bucket_name}")
+            lifecycle_ui.navigate_to_bucket_lifecycle(bucket_name)
+
+        rules = lifecycle_ui.get_lifecycle_rules_list()
+
+        if not rules:
+            logger.info("No existing rules found, creating a rule to delete")
+            rule_to_delete = create_unique_resource_name("rule", "to-delete")
+            lifecycle_ui.create_lifecycle_rule(
+                rule_name=rule_to_delete,
+                scope="whole_bucket",
+                rules={"expiration": {"days": 30}},
+            )
+
+            lifecycle_ui.do_click(lifecycle_ui.bucket_tab["management_tab"])
+            time.sleep(2)
+
+            rules = lifecycle_ui.get_lifecycle_rules_list()
+            assert rule_to_delete in rules, f"Failed to create rule {rule_to_delete}"
+        else:
+            rule_to_delete = rules[0]
+            logger.info(f"Found existing rule to delete: {rule_to_delete}")
+
+        initial_rule_count = len(rules)
+        logger.info(f"Initial rule count: {initial_rule_count}")
+
+        lifecycle_ui.delete_lifecycle_rule(rule_to_delete)
+        time.sleep(3)  # Wait for UI to update
```

Comment:
Because this operation first has to be processed by the backend, we might need to wait longer than 3 seconds in this case. Please increase the sleep or use a TimeoutSampler or retry.